### PR TITLE
Iceoryx - Add 2.0.0

### PR DESCRIPTION
**Pull request by dotChris90**
Specify library name and version:  iceoryx/2.0.0

Added the 2.0.0 release

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SSE4** [comment]
[shm_*](https://man7.org/linux/man-pages/man3/shm_open.3.html) functions require `rt` library

**Comment by SSE4** [comment]
seems like it need CMake 3.16+ as a build requirement (CI uses 3.15 due to the tribe policy)

**Comment by dotChris90** [comment]
@SSE4  thanks for telling me the reason behind! :D i added build_requirement!

**Comment by dotChris90** [comment]
@uilianries I will init the logger of iceoryx simple and see if include header and linking libs.  :D

**Review by SSE4 — DISMISSED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

File: recipes/iceoryx/all/conanfile.py [hunk 1]
```diff
@@ -52,6 +52,9 @@ def requirements(self):
         if self.settings.os == "Linux":
             self.requires("acl/2.3.1")
 
+    def build_requirements(self):
+        self.tool_requires("cmake/3.16.2")
```
Comment (jgsogo) [inline recipes/iceoryx/all/conanfile.py:56]: Is this required for all the versions? Only for new `2.0.0`?

**Review by jgsogo — COMMENTED** [review]
It is better if we add the build-requires just for the versions that need it. It will save some bandwidth while building all the configurations.

**Review by SSE4 — APPROVED** [review]

**Review by jgsogo — APPROVED** [review]
Thanks for simplifying the `test_package` 👍

**Comment by dotChris90** [comment]
> Member

no issue ;)

**Review by uilianries — APPROVED** [review]

**Pull request merged**